ui/src/api/mixins/ApiResource.ts

```diff
@@ -0,0 +1,24 @@
+import { Constructor } from '@tpluscode/rdfine'
+import { Mixin } from '@tpluscode/rdfine/lib/ResourceFactory'
+
+declare module 'alcaeus/Resources' {
+  export interface HydraResource {
+    clientPath: string
+  }
+}
+
+export function apiResourceMixin (rootUrl: string, separator: string): Mixin {
+  function ApiResourceMixin<Base extends Constructor> (base: Base) {
+    class BaseClass extends base {
+      get clientPath () {
+        return this.id.value.replace(rootUrl, '').replace('/', separator)
+      }
+    }
+
+    return BaseClass
+  }
+
+  ApiResourceMixin.shouldApply = true
+
+  return ApiResourceMixin
+}
```

Comment:
Hydra resource objects are extended using partial mixin classes which are added to the prototype chain.

The `shouldApply` can also be a function and the preferred way is to associate a mixin function with an RDF type like `ApiResourceMixin.appliesTo = _NamedNode_`